src/monitoring/index.ts

```diff
@@ -0,0 +1,295 @@
+}
+
+interface KeyCdnResponse {
+  data: {
+    zonealiases: keyCdnZoneAlias[]
+  }
+}
+
+interface RedirectionDomain {
+  source: string
+  target: string
+}
+
+interface ReportCard {
+  domain: string
+  type: typeof IsomerHostedDomainType[keyof typeof IsomerHostedDomainType]
+  aRecord: string[]
+  quadArecord: string[]
+  cNameRecord: string[]
+  caaRecord: string[]
+}
+
+function isKeyCdnZoneAlias(object: unknown): object is keyCdnZoneAlias {
+  return "name" in (object as keyCdnZoneAlias)
+}
+
+function isKeyCdnResponse(object: unknown): object is KeyCdnResponse {
+  return "data" in (object as KeyCdnResponse)
+}
+
+export default class MonitoringService {
+  private readonly launchesService: MonitoringServiceInterface["launchesService"]
+
+  private readonly monitoringServiceLogger = parentLogger.child({
+    module: "monitoringService",
+  })
+
+  constructor({ launchesService }: MonitoringServiceInterface) {
+    autoBind(this)
+    this.launchesService = launchesService
+  }
+
+  getKeyCdnDomains() {
+    const keyCdnApiKey = config.get("keyCdn.apiKey")
+
+    return ResultAsync.fromPromise(
+      fetch(`https://api.keycdn.com/zonealiases.json`, {
+        headers: {
+          Authorization: `Basic ${btoa(`${keyCdnApiKey}:`)}`,
+        },
+      }),
+      (error) => new MonitoringError(`Failed to fetch zones: ${error}`)
+    )
+      .andThen((response) => {
+        if (!response.ok) {
+          return errAsync(
+            new MonitoringError(
+              `Failed to retrieve zones: ${response.statusText}`
+            )
+          )
+        }
+        return okAsync(response)
+      })
+      .andThen((response) =>
+        ResultAsync.fromPromise(
+          response.json(),
+          (error) => new MonitoringError(`Failed to parse response: ${error}`)
+        )
+      )
+      .andThen((data: unknown) => {
+        if (!isKeyCdnResponse(data)) {
+          return errAsync(new MonitoringError("Failed to parse response"))
+        }
+
+        const domains = data.data.zonealiases
+          .filter(isKeyCdnZoneAlias)
+          .map((zone: keyCdnZoneAlias) => zone.name)
+          .map(
+            (domain) =>
+              ({
+                domain,
+                type: IsomerHostedDomainType.KEYCDN,
+              } as IsomerHostedDomain)
+          )
+
+        return okAsync(domains)
+      })
+  }
+
+  getAmplifyDeployments() {
+    return this.launchesService.getAllDomains().map((domains) =>
+      domains.map(
+        (domain) =>
+          ({
+            domain,
+            type: IsomerHostedDomainType.AMPLIFY,
+          } as IsomerHostedDomain)
+      )
+    )
+  }
+
+  /**
+   * While most of our redirections are in our DB, we do have ad-hoc redirections.
+   * @returns List of redirection domains that are listed in the isomer-redirection repository
+   */
+  getRedirectionDomains() {
+    const SYSTEM_GITHUB_TOKEN = config.get("github.systemToken")
+    const OctokitRetry = Octokit.plugin()
```

Comment:
going off docs [here](https://github.com/octokit/plugin-retry.js/#readme) 

```ts
const MyOctokit = Octokit.plugin(**retry**);
const octokit = new MyOctokit({ auth: "secret123" });

// retries request up to 3 times in case of a 500 response
octokit.request("/").catch((error) => {
  if (error.request.request.retryCount) {
    console.log(
      `request failed after ${error.request.request.retryCount} retries`,
    );
  }

  console.error(error);
});```

Reply:
just wanted to nudge on this! if we care about retries, we might want to add the plugin in!

Reply:
this has been modified